python/ray/_raylet.pyx

```diff
@@ -702,3 +710,15 @@ cdef class CoreWorker:
         return ProfileEvent.make(
             self.core_worker.get().CreateProfileEvent(c_event_type),
             extra_data)
+
+    def deserialize_actor_handle(self, const c_string &bytes):
+        c_actor_id = self.core_worker.get().DeserializeActorHandle(bytes)
+        actor_id = ActorID(c_actor_id.Binary())
+        return actor_id
+
+    def serialize_actor_handle(self, ActorID actor_id):
+        cdef:
+            CActorID c_actor_id = actor_id.native()
+            c_string output
+        self.core_worker.get().SerializeActorHandle(c_actor_id, &output)
```

Comment:
Is it necessary to call into core worker and get the handle? Seems like asking for a RAY_CHECK failure due to some unforeseen serialization path.

Reply:
The point is to not expose actor handles to Python anymore. What is the alternative you're thinking of?

Also, it is definitely a bug if you try to serialize an actor handle that you don't have. This is if you generated an actor ID out of band and tried to get the actor handle to it. To prevent crashes, we can return an error here if the handle doesn't exist. Would that work?

Reply:
That's one advantage of lazy registration when you first use the handle right, it would handle all these cases automatically without crashing?

In any case, I don't think RAY_CHECK is the right choice for an error that can be triggered by application level code.

Reply:
Er...I'm not sure how lazy registration would help this situation. Can you explain more?

I agree, but I think it's out of scope for this PR. We can start a discussion on how to do this separately.